addon/components/step-manager.ts

```diff
@@ -64,6 +64,11 @@ export default class StepManagerComponent extends Component {
    */
   currentStep: StepName | undefined;
 
+  /**
+   * Callback action to be triggered when the current step changes.
+   */
+  onTransition?: Function;
```

Comment:
This seems fine, although we can also be slightly more specific with a type signature like this:

```typescript
onTransition?: (name: StepName) => void
```

since it denotes the specific case that we'll pass a `StepName` and don't care about the return value.